src/ceph_fuse.cc

```diff
@@ -202,10 +202,21 @@ int main(int argc, const char **argv, const char *envp[]) {
 	ceph_assert(ver != 0);
         bool client_try_dentry_invalidate = g_conf().get_val<bool>(
           "client_try_dentry_invalidate");
-	bool can_invalidate_dentries =
+        bool can_invalidate_dentries =
           client_try_dentry_invalidate && ver < KERNEL_VERSION(3, 18, 0);
-	auto test_result = client->test_dentry_handling(can_invalidate_dentries);
-	int tr = test_result.first;
+        uint64_t max_retries = g_conf().get_val<uint64_t>(
+          "client_max_retries_on_remount_failure");
+        std::pair<int, bool> test_result;
+        uint64_t i = 0;
+        int tr;
+        do {
+          test_result = client->test_dentry_handling(can_invalidate_dentries);
+          tr = test_result.first;
+	  if (tr) {
+            sleep(1);
+	  }
+        } while (i++ < max_retries && tr);
+
```

Comment:
Yeah, right. I will fix it.

Reply:
If we are not considering the first run as a retry, then I suppose it should be `while ( i++ < max_retries && tr)` or `while (++i <= max_retries && tr)`, otherwise we are missing one attempt to retry.

Reply:
The `client_max_retries_on_remount_failure`:
```
desc: number of consecutive failed remount attempts for invalidating kernel dcache
    after which client would abort.
```
IMO it should include the first try.

Reply:
Okay